### PR TITLE
[TRTLLM-5812][feat] support FP8 row-wise dense GEMM in torch flow

### PR DESCRIPTION
**Pull request by DylanChen-NV**
# [TRTLLM-5812][feat] support FP8 row-wise dense GEMM in torch flow

## Description

Adapter existing FP8 row-wise dense GEMM kernels (sm89/sm90/sm120) to torch workflow.
Support FP8 row-wise GEMM as a TunableRunner for better performance.

## Test Coverage

tests/unittest/_torch/thop/test_fp8_rowwise_linear.py: Test the output correctness of row-wise torch op

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by Tracin — APPROVED** [review]

File: tensorrt_llm/_torch/modules/linear.py [hunk 1]
```diff
@@ -390,6 +390,107 @@ def load_weights_fused_gate_up_linear(self, module: Linear,
...
+            module, weights)
+        fused_weight = torch.cat((gate_weight, up_weight))
+        copy_weight(module.weight, fused_weight)
+
+        scale_name = self._get_scale_name(weights)
+        left_scale = load_weight_shard(weights[0][scale_name], module.tp_size,
+                                       module.tp_rank, module.tp_mode)
+        right_scale = load_weight_shard(weights[1][scale_name], module.tp_size,
+                                        module.tp_rank, module.tp_mode)
+        fused_scale = torch.cat([left_scale, right_scale], dim=0)
```
Comment (tomeras91) [inline tensorrt_llm/_torch/modules/linear.py:490]: nit: this `cat` operation is not consistent with the ones in `load_weights_fused_qkv_linear` - it uses a list of tensors and specifies the dim, and the ones in that function use a tuple of tensors and don't specify dim. Can you improve consistency here?

**Review by tomeras91 — COMMENTED** [review]
Overall LGTM. Added a few nit comments.

What's more important: Can you edit the PR description, remove the irrelevant info from the description template and add important information on what is added in this PR?

File: tensorrt_llm/_torch/modules/linear.py (hunk 1 above)
Comment (DylanChen-NV) [reply tensorrt_llm/_torch/modules/linear.py]: Sure, that would be better. Updated.

**Review by tomeras91 — APPROVED** [review]

**Review by byshiue — APPROVED** [review]
